plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -525,7 +529,7 @@ open class DefaultPageCreator(
                     .groupBy { it.name } // This groupBy should probably use LocationProvider
                     // This hacks displaying actual typealias signatures along classlike ones
                     .mapValues { if (it.value.any { it is DClasslike }) it.value.filter { it !is DTypeAlias } else it.value }
-                    .entries.sortedBy { it.key }
+                    .entries.sortedWith(groupKeyComparator)
```

Comment:
Could you please add a unit test to check the described use case as well?

Here it might be a bit more difficult because of the deeply nested groupings. The test class is `PageNodeMergerTest`. You can have a look at the test named `should not ignore case when grouping by name` - you'll probably need something similar